marvin-ai/marvin-engine-executor
# Json schema bug

Hey guys @takabayashi @lucasbm88 , my last PR about Json schema bug fix #41 did not solve all those problems, @takabayashi had to eliminate $ref block from main json schema for successful merge.

This PR i have changed the Json Validator, and looks like it worked. I need you guys to run some tests to confirm this.

Test steps:
1 - Choose your engine => workon engine-name-env
2 - Bring up  engine http server => marvin engine-httpserver -e /path/to/your/jar/marvin-engine-executor-assembly-0.0.1.jar
3 - No Exception should be thrown

Another Test:
1 - Open your engine.metadata file and change some properties (number to string or vice versa)
2 - choose your engine => workon engine-name-env
3 - Bring up  engine http server => marvin engine-httpserver -e /path/to/your/jar/marvin-engine-executor-assembly-0.0.1.jar
4 - Marvin log should show metadata error detail and MarvinEExecutorException should be thrown with message: Invalid engine metadata file. Check your engine metadata file.

Feel free to run anothers tests too, tks!
